interview-questions.md

```diff
@@ -7,53 +7,170 @@ Then, research the question to expand on your answer. Even if you feel you have
+ that has an actual value of an attribute. The rows in the table represent a collection of related values 
+ Each row in a table could be marked with a unquie idenifier, a primayr key. 
 
+ Examples: Microsoft SQL Server, Oracle Database, MySQL, PostgreSQL and IBM Db2
 
+Non-relational databases are those that do not use SQL as their primary data access language and are referred to as NoSQL databases.
+Because data in a NoSQL database does not have to comply to a pre-defined schema, unlike data in relational databases, these databases are ideal for storing unstructured or semi-structured data.
+One benefit of NoSQL databases is that they allow developers to make changes to the database on the fly without disrupting apps that use it.
 
-5. What are primary keys? Why are they important?
+ Examples: Apache Cassandra, MongoDB, CouchDB, and CouchBase
 
-  Your answer:
+Any database that is built to function in the cloud is referred to as a cloud database.
+Cloud databases, like other cloud-based applications, provide flexibility, scalability, and high availability.
+Because many cloud databases are provided as a service, they are frequently low-maintenance.
 
-  Researched answer:
 
+Examples: Microsoft Azure SQL Database, Amazon Relational Database Service, Oracle Autonomous Database.
+
+
+5. What are primary keys? Why are they important?
+
+  Your answer: Primary Keys are the listed in the column and important because they contain the values. 
+  For example Titles would First Name, Last Name, DOB would be primary keys in a database. 
+  When I worked for a univerity, the student ID number was a primary key in our database. 
+  
+  Researched answer:
+  In a database a primary key of a relational table uniquly identifies each record in the table. Databases use keys to compare, 
+  sort and store records and to create relationships between records. Primary key are essential to database because these keys 
+  are used to compare and otherwise define relationships between individual records. Never change the primary key after you assign it. 
 
 
 ## Looking Ahead: Terms for Next Week
 Research and define the following terms to the best of your ability.
 
-1. RESTful routes:
+1. RESTful routes: A RESTful route is one that maps HTTP verbs (get, post, put, delete, patch) to CRUD activities in the controller 
+(create, read, update, delete).Instead of depending just on the URL to determine which site to visit, a RESTful route 
+uses both the HTTP verb and the URL to choose which site to visit. 
+
+2. JSON:JSON (JavaScript Object Notation) is a data-exchange format that is simple to use. 
+Reading and writing are simp>
+trustedkey <%= @keys_trusted.join(' ') %>
+<% end -%>
+<% if @keys_requestkey != '' -%>
+requestkey <%= @keys_requestkey %>
+<% end -%>
+<% if @keys_controlkey != '' -%>
+controlkey <%= @keys_controlkey %>
+<% end -%>
```

Comment:
Not sure if github is doing something funny here, but I don't think lines 131 - 139 belong here.